### ISSUE TITLE
Error when directory contains too much directories and files

### ISSUE THREAD

**Issue by andyyehoo** [solidiquis/erdtree#2]
$ erdtree -l 1

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/file_tree/tree_node.rs:48:51
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by solidiquis** [comment]
Interesting. I am unable to replicate this error (even in an empty directory). Would you mind sharing what your directory structure looks like?

**Comment by andyyehoo** [comment]
> 

This needs to happen in a very large directory, with many sub directories and files. It will not happen in a empty directory

**Comment by solidiquis** [comment]
@andyyehoo thank you for bringing this to my attention.

There were two problems:
1. Files without read permissions panicked when I tried to extract its metadata.
2. Symlinks that point to non-existent files would also panic when trying to access its metadata.

I patched this up with f6803e081929789d75f1974110c3c22cfa7ad87b and just pushed a new 0.0.2 tag.

Closing this issue. Thank you for bringing this to my attention and let me know if there are any other problems.
